ext/random/randomizer.c

```diff
@@ -292,14 +293,48 @@ PHP_METHOD(Random_Randomizer, getBytes)
 	size_t length = (size_t)user_length;
 	retval = zend_string_alloc(length, 0);
 
+	php_random_result result;
+	while (total_size + 8 <= length) {
+		result = engine.algo->generate(engine.state);
+		if (EG(exception)) {
+			zend_string_free(retval);
+			RETURN_THROWS();
+		}
+
+		/* If the result is not 64 bits, we can't use the fast path and
+		 * we don't attempt to use it in the future, because we don't
+		 * expect engines to change their output size.
+		 *
+		 * While it would be possible to always memcpy() the entire output,
+		 * using result.size as the length that would result in much worse
+		 * assembly, because it will actually emit a call to memcpy()
+		 * instead of just storing the 64 bit value at a memory offset.
+		 */
+		if (result.size != 8) {
+			goto non_64;
+		}
+
+#ifdef WORDS_BIGENDIAN
+		uint64_t swapped = ZEND_BYTES_SWAP64(result.result);
+		memcpy(ZSTR_VAL(retval) + total_size, &swapped, 8);
+#else
+		memcpy(ZSTR_VAL(retval) + total_size, &result.result, 8);
+#endif
+		total_size += 8;
+	}
+
```

Comment:
I am confused, is it intended for this to run the second while loop again? Or should it skip it to return the value directly?

Reply:
It will enter the second loop if the requested length is not a multiple of 8 to handle the remaining 1-7 bytes.